File: CodeForces/501-600/546/A/546A.go
// Problem 546A: Soldier and Bananas
// https://codeforces.com/contest/546/problem/A
// Executed with: go version go1.24.4 linux/arm64
// Linux version: Linux kernel 6.17.0-5-generic
// Submitted on: October 26th, 2025
// Codeforces language used: Go 1.22.2

package main

import (
	"fmt"
)

func main() {
	var k, n, w int
	fmt.Scan(&k, &n, &w)

	var total_cost_multi = ((w * (w + 1)) / 2)
	var total_cost = k * total_cost_multi
	var amount_to_borrow = total_cost - n

	if amount_to_borrow > 0 {
		fmt.Println(amount_to_borrow)
	} else {
		fmt.Println(0)
	}
}

File: CodeForces/501-600/546/A/546A.php
<?php

// Problem 546A: Soldier and Bananas
// https://codeforces.com/contest/546/problem/A
// Executed with: PHP 8.4.11 (cli) (built: Aug 13 2025 01:43:48) (NTS)
// Linux version: Linux kernel 6.17.0-5-generic
// Written on: October 27th, 2025
// Codeforces language used: Cannot be submitted

$arr = array_map('intval', explode(' ', trim(fgets(STDIN))));
$k = $arr[0];
$n = $arr[1];
$w = $arr[2];

$total_cost_multi = (($w * ($w + 1)) / 2);
$total_cost = $k * $total_cost_multi;
$amount_to_borrow = $total_cost - $n;

echo max(0, $amount_to_borrow);
echo "\n";

?>
